examples/tools/scene_viewer.rs

```diff
@@ -0,0 +1,407 @@
+            // a Sphere, and then back to an Aabb to find the conservative min and max points.
+            let sphere = Sphere {
+                center: transform.mul_vec3(aabb.center),
+                radius: (transform.scale * aabb.half_extents).length(),
+            };
+            let aabb = Aabb::from(sphere);
+            min = min.min(aabb.min());
+            max = max.max(aabb.max());
+        }
+
+        let size = (max - min).length();
+        let aabb = Aabb::from_min_max(min, max);
+
+        if !scene_handle.has_camera {
+            let transform =
+                Transform::from_translation(aabb.center + size * Vec3::new(0.5, 0.25, 0.5))
+                    .looking_at(aabb.center, Vec3::Y);
+            let view = transform.compute_matrix();
+            let mut perspective_projection = PerspectiveProjection::default();
+            perspective_projection.far = perspective_projection.far.max(size * 10.0);
+            let view_projection = view.inverse() * perspective_projection.get_projection_matrix();
+            let frustum = Frustum::from_view_projection(
+                &view_projection,
+                &transform.translation,
+                &transform.back(),
+                perspective_projection.far(),
+            );
+
+            info!("Spawning a 3D perspective camera");
+            commands.spawn_bundle(PerspectiveCameraBundle {
+                camera: Camera {
+                    near: perspective_projection.near,
+                    far: perspective_projection.far,
+                    ..default()
+                },
+                perspective_projection,
+                frustum,
+                transform,
+                ..PerspectiveCameraBundle::new_3d()
+            });
+
+            scene_handle.has_camera = true;
+        }
+
+        if !scene_handle.has_light {
+            // The same approach as above but now for the scene
+            let sphere = Sphere {
+                center: aabb.center,
+                radius: aabb.half_extents.length(),
+            };
+            let aabb = Aabb::from(sphere);
+            let min = aabb.min();
+            let max = aabb.max();
+
+            info!("Spawning a directional light");
+            commands.spawn_bundle(DirectionalLightBundle {
+                directional_light: DirectionalLight {
+                    shadow_projection: OrthographicProjection {
+                        left: min.x,
+                        right: max.x,
+                        bottom: min.y,
+                        top: max.y,
+                        near: min.z,
+                        far: max.z,
+                        ..default()
+                    },
+                    shadows_enabled: false,
+                    ..default()
+                },
+                ..default()
+            });
+
+            scene_handle.has_light = true;
+        }
+    }
+}
+
+// TODO: Register all types in CameraController so that it can be registered
```

Comment:
Should this TODO comment still be here?

Reply:
Yes. I've worked around it in this PR. I could register the KeyCode type and any other necessary types, it just seems wrong to have to keep doing this in the engine. Probably just all component/resource types should be registered always.

Reply:
Hmm, although, the code has changed a bit since I originally tried to add the CameraController component to the first entity with a Camera directly into the Scene's World and now I have to wait for the Scene to have fully loaded into the main World before doing things, so I think I will leave it like this and remove the TODO.